build/config/BUILDCONFIG.gn

```diff
@@ -303,6 +303,7 @@ _native_compiler_configs = [
   "//build/config:feature_flags",
   "//build/config/compiler:compiler",
   "//build/config/compiler:cxx_version_default",
+  "//build/config/compiler:objc_arc",
```

Comment:
IMO it would be much clearer if the config contained no platform conditions, and it was added here in a platform condition as with windows configs a few lines later. Locally, this looks like a bug without going and looking at the impl, and conversely the impl containing a check seemed surprising since I wouldn't expect a config with `objc` in the name to be used except on macOS and iOS.

Reply:
+1 Seems like a reasonable approach. 

And if we ever get a GNUStep implementation (and if GNUStep then gets ARC support) and have to adjust the ios/macos condition, I think we'll be able to claim victory.